packages/rsocket-tcp-client/src/TcpDuplexConnection.ts

```diff
@@ -0,0 +1,87 @@
+import net from "net";
+import { Deferred } from "@rsocket/rsocket-core";
+import {
+  deserializeFrameWithLength,
+  DuplexConnection,
+  FlowControlledFrameHandler,
+  Frame,
+  serializeFrameWithLength,
+} from "@rsocket/rsocket-types";
+
+export class TcpDuplexConnection extends Deferred implements DuplexConnection {
+  private handler: FlowControlledFrameHandler;
+  private error: Error;
+
+  constructor(private socket: net.Socket) {
+    super();
+
+    socket.on("close", this.handleClosed.bind(this));
+    socket.on("error", this.handleError.bind(this));
+    socket.on("data", this.handleData.bind(this));
+  }
+
+  handle(handler: FlowControlledFrameHandler): void {
+    if (this.handler) {
+      throw new Error("Handle has already been installed");
+    }
+
+    this.handler = handler;
+  }
+
+  get availability(): number {
+    throw new Error("Method not implemented.");
+  }
+
+  close(error?: Error) {
+    if (this.done) {
+      return;
+    }
+
+    this.socket.removeListener("close", this.handleClosed.bind(this));
+    this.socket.removeListener("error", this.handleError.bind(this));
+    this.socket.removeListener("message", this.handleData.bind(this));
+
+    this.socket.destroy(error);
+
+    delete this.socket;
+
+    super.close(error);
+  }
+
+  send(frame: Frame): void {
+    if (this.done) {
+      return;
+    }
+
+    const buffer = serializeFrameWithLength(frame);
+
+    // if (!this._socket) {
+    //   throw new Error(
+    //     "TcpDuplexConnection: Cannot send frame, not connected."
+    //   );
+    // }
```

Comment:
This pattern was present from the WebSocket impl. Not sure we need to cover this case? In what scenario would `send()` be called where `socket` is null?

Reply:
I dont think it was a good pattern there. Better to drop frame since eventually it should be closed in any case